lab_6_pipeline/pipeline.py

```diff
@@ -144,16 +229,89 @@ def __init__(self, corpus_manager: CorpusManager):
+                                  'PR': 'ADP',
+                                  'CONJ': 'CCONJ',
+                                  'PART': 'PART',
+                                  'INTJ': 'INTJ'}}
+            with open(mapping_file, 'w', encoding='utf-8') as json_file:
+                json.dump(ud_mapping, json_file)
+        self._tag_converter = MystemTagConverter(mapping_file)
 
     def _process(self, text: str) -> List[ConlluSentence]:
         """
         Returns the text representation as the list of ConlluSentence
         """
+        sentences = split_by_sentence(text)
+        conllu_sentences = []
+        for idx, sent in enumerate(sentences):
+            sent_analyzed = pymystem3.Mystem().analyze(sent)
+            conllu_wordlist = []
+            index = 1
+            for word in sent_analyzed:
+                try:
+                    tags_list = re.findall(r'\w+', word['analysis'][0]['gr'])
+                except KeyError:
+                    if re.findall(r'\s+', word['text']):
+                        continue
+                    conllu_token = ConlluToken(word['text'])
+                    conllu_token.set_position(index)
+                    index += 1
+                    if patterns := re.findall(r'\d+', word['text']):
+                        conllu_token.set_morphological_parameters(
+                            MorphologicalTokenDTO(patterns[0], 'NUM'))
+                    elif patterns := re.findall(r'\.', word['text']):
+                        conllu_token.set_morphological_parameters(
+                            MorphologicalTokenDTO(patterns[0], 'PUNCT'))
+                    conllu_wordlist.append(conllu_token)
+                    continue
+                except IndexError:
+                    if patterns := re.findall(r'[A-Za-z]+', word['text']):
+                        conllu_token = ConlluToken(patterns[0])
+                        conllu_token.set_position(index)
+                        index += 1
+                        conllu_token.set_morphological_parameters(
+                            MorphologicalTokenDTO(patterns[0], 'X'))
+                        conllu_wordlist.append(conllu_token)
+                        continue
+                    continue
+                ud_tag = self._tag_converter.convert_pos(tags_list[0])
+                conllu_token = ConlluToken(word['text'])
+                conllu_token.set_position(index)
+                index += 1
+                conllu_token.set_morphological_parameters(
+                    MorphologicalTokenDTO(word['analysis'][0]['lex'], ud_tag))
+                conllu_wordlist.append(conllu_token)
+            conllu_sentences.append(ConlluSentence(idx, sent, conllu_wordlist))
+        return conllu_sentences
 
     def run(self) -> None:
         """
         Performs basic preprocessing and writes processed text to files
         """
+        articles = self._corpus.get_articles()
+        for id in articles:
```

Comment:
iterate through keys and values as items simultaneously

Reply:
Done